File: app/src/main/java/com/savvasdalkitsis/gameframe/injector/infra/network/InterceptorInjector.java
package com.savvasdalkitsis.gameframe.injector.infra.network;

import com.savvasdalkitsis.gameframe.ip.network.IpBaseHostInterceptor;

import okhttp3.Interceptor;

import static com.savvasdalkitsis.gameframe.injector.ip.repository.IpRepositoryInjector.ipRepository;

public class InterceptorInjector {

    static Interceptor ipBaseHostInterceptor() {
        return new IpBaseHostInterceptor(ipRepository());
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/injector/infra/network/InterceptorInjector.kt
package com.savvasdalkitsis.gameframe.injector.infra.network

import com.savvasdalkitsis.gameframe.injector.feature.ip.repository.IpRepositoryInjector.ipRepository
import com.savvasdalkitsis.gameframe.feature.ip.network.IpBaseHostInterceptor
import okhttp3.Interceptor

object InterceptorInjector {

    internal fun ipBaseHostInterceptor(): Interceptor = IpBaseHostInterceptor(ipRepository())
}
